main.nf

```diff
@@ -203,6 +220,53 @@ process get_software_versions {
     """
 }
 
+/*
+ * Generate BAM files if input files are CRAM files
+ */
+
+if ( params.cram_files ) {
+
+  if ( params.reference_fasta ) {
+
+    process cramToBamWithReference {
+      tag "$name"
+      label 'process_medium'
+
+      input:
+      set val(name), file(cram) from ch_cram_files
+      path fasta from params.reference_fasta
```

Comment:
I tried setting the `fasta` input conditionally, but I couldn't get it to work, so I'm conditionally including the process. Let me know if there's a more elegant approach here.

Reply:
Never mind. I stumbled across an [approach](https://github.com/nf-core/sarek/blob/2.7.1/main.nf#L2037) in Sarek, which I was able to use here to de-duplicate the processes.